backdoor/src/main/java/org/codice/acdebugger/backdoor/Backdoor.java

```diff
@@ -283,9 +285,15 @@ private String getBundle0(@Nullable Object obj) {
       return ((BundleContext) obj).getBundle().getSymbolicName();
     } else if (obj instanceof ProtectionDomain) {
       // check if we have a protection domain with Eclipse's permissions
-      final Object permissions = ((ProtectionDomain) obj).getPermissions();
-
-      if (permissions instanceof BundlePermissions) {
+      final PermissionCollection permissions = ((ProtectionDomain) obj).getPermissions();
+
+      // we cannot reference org.eclipse.osgi.internal.permadmin.BundlePermissions directly as it is
+      // not exported by Eclipse
+      if ((permissions != null)
+          && permissions
+              .getClass()
+              .getName()
+              .equals("org.eclipse.osgi.internal.permadmin.BundlePermissions")) {
```

Comment:
✏️ Should make this a constant.

Reply:
I would rather keep it inline like this to make it clearer. I would have preferred `instanceof` but that is an internal class in Eclipse so 😞